### PR TITLE
ACM-6165: Include property with list of relatedUids in related objects

### PR DESCRIPTION
**Pull request by SherinV**
### Related Issue
https://issues.redhat.com/browse/ACM-6165

### Description of changes
- Related objects include a new property `_relatedUids` with a list of objects it is related to.

### CODE REVIEW

**Review by jlpadilla — CHANGES_REQUESTED** [review]
@philipwu08 @SherinV I want to replace the words "parent" and "child" in this PR.  Not all relations are parent-child, it is only one type of relationship.
To avoid confusion, let's use something more generic like `relatedUid`.

**Comment by SherinV** [comment]
> 

Replaced

File: pkg/resolver/related.go [hunk 1]
```diff
@@ -342,6 +357,39 @@ func (s *SearchResult) updateKindMap(uid string, kind string, levelMap map[strin
 	levelMap[kind] = uids
 }
 
+// map the related result uids to the uids in the search input
```
Comment (jlpadilla) [inline pkg/resolver/related.go:360]:
```suggestion
// Maps the related result uids to the uids in the search result set.
```

File: pkg/resolver/related.go [hunk 2]
```diff
@@ -342,6 +357,39 @@ func (s *SearchResult) updateKindMap(uid string, kind string, levelMap map[strin
...
+				}
+			}
+			klog.V(9).Infof("uid %s is  related to uids %+v.", resultUid, resultToCurrSearchUidsMap[resultUid])
+			// Need to process only one uid in the path as it is either the sourceid or the destid that can be related
+			break
+		}
+	}
+}
+
+func checkIfInArray(lookupMap []string, uid string) bool {
```
Comment (jlpadilla) [inline pkg/resolver/related.go:384]: Just FYI, go 1.21 will add a new standard package `slices` to simplify this. It also has a `slices.Sort()` which we have somewhere else in this project.
https://pkg.go.dev/slices

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

**Pull request merged**